js/src/components/contact-information/phone-number-card/useCountdown.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useCallback, useRef } from '@wordpress/element';
+
+/**
+ * @typedef {Object} Countdown
+ * @property {number} second The current remaining seconds of countdown.
+ * @property {number} callCount The calling count of `startCountdown` function.
+ * @property {(countdownSecond: number) => void} startCountdown Start the countdown with the given starting seconds.
+ */
+
+/**
+ * Get the corresponding countdown states and its start function by given `handle`.
+ * The states would be updated every 1000 ms interval till counting down to 0.
+ *
+ * @param {string} handle The countdown handle.
+ * @return {Countdown} The corresponding countdown states and its start function.
+ */
+export default function useCountdown( handle ) {
```

Comment:
❔ Not sure if you are aware about this, might be helpful: https://react-hooks.org/docs/usecountdown/

Think we can consider using that library and we don't need to reinvent the wheel and don't need to come out with our tests.

Reply:
Thanks! I didn't know it. After looking into its doc, I think it's not so suitable for our use case. If we apply it, we have to control the `endTime` state and record the calling count of the countdown timer from outside, or still need to create a custom hook to wrap additional states with that `useCountdown` hook. Consider the implementation of the current `useCountdown` is short and simple, I think we can implement it instead of introducing a new dependency.